odoo/addons/base/migrations/11.0.1.3/pre-migration.py

```diff
@@ -24,6 +24,39 @@
     ('ir.actions.report.xml', 'ir.actions.report')
 ]
 
+_obsolete_tables = (
+    "base_action_rule",
+    "base_action_rule_lead_test",
+    "base_action_rule_line_test",
+    "crm_activity",
+    "hr_timesheet_sheet_sheet",
```

Comment:
```suggestion
```
This table is renamed to 'hr_timesheet_sheet' in `hr_timesheet_sheet` module. I think tables like this one shouldn't be in this list.

Reply:
Basically, I think you should delete from this list:

base_action_rule
base_action_rule_lead_test
base_action_rule_line_test
hr_timesheet_sheet_sheet
stock_picking_wave

Not sure if I left something.

Reply:
What do you think @pedrobaeza ?

Reply:
@Yajo if other module defines the same m2o/m2m field, the FK is re-added by ORM. If so, then I think full list is correct and if you install for example `hr_timesheet_sheet` module, everything will be set as it was. If not, you don't have a table with FKs consuming resources when doing operations in linked tables.

Reply:
Yes, the only problem would be pre-migration scripts, if any, because they could rely on a missing FK.

I think I'm just gonna remove the last commit, as this PR is drifting out of scope.

Reply:
I don't think that's the case, and I prefer to remove this by default and if any pre-migration should rely on that, then re-add it (which I don't think there's any case). Removing this, we are losing potentially performance for a non clear need of restoration.

Reply:
If this were the 1st commit in this migration, yes... But to be sure we're not breaking anything here, I should:

1. Check which modules defined these tables in the previous version.
1. Check if those are properly migrated in this version.
1. Check no pre-migration script relies on an FK for these tables.
1. Remove the tables from the list or fix the affected premigration scripts.

So, when this PR started as a simple mechanism to improve performance of the procurement migration, it would become a complex monster that would potentially affect the whole migration process.

I agree it would be good to drop all those FKs, but at the point where we're at, and with the implied risk of introducing new bugs, let me insist that we stay pragmatic and just fix the issue at hand, please.

Reply:
I save you all the time: you can do it without risk. I can assure you, so let's go this way.